consoles/virtio_screen.pm

```diff
@@ -264,6 +264,37 @@ sub read_until {
     return {matched => 1, string => $overflow . $prematch . $match};
 }
 
+=head2 peak
+
+Read and return pending data without consuming it. This is useful if you are
+about to destroy the virtio_screen instance, but want to keep any pending
+data. However this does not wait for any data in particular so this races with
+the backend and data transport. Therefor it should only be used when there is
+no information available about what data is expected to be available.
+
+=cut
+sub peak {
+    my ($self, %nargs) = @_;
+    my $buflen     = $nargs{buffer_size} || 4096;
+    my $total_read = 0;
+    my $buf        = '';
+    my $read;
+
+    bmwqemu::log_call(%nargs);
+  LOOP: {
```

Comment:
@richiejp, @mudler  I wonder if it's possible to do this without a GOTO :P

Reply:
The problem is Perl does not have a do-while loop (the `do` keyword has some other meaning). So we can either unroll the first iteration of the loop so that it works with `while` or do something else, but I am not sure what. I think unrolling the loop looks pretty ugly.

Reply:
You can actually ```do { } while (..)``` or either ```do {} unless (...)``` even if ```do``` have different meanings. But i do think it's maybe part of a follow up and the refinement should not be blocking

Reply:
I don't think you can use `last` inside the `do` body.

Reply:
you can, but Perl will do something you are not expecting :) ( as this is just about 'style' talks and a bit subjective) i usually tend up to break the loops in the while condition, or change the loop style based on what is the purpose (``` until ( <cond > ) { .... } ``` etc.) but it's a matter of taste

Reply:
you can enclose the body with curly braces:  `{ do my_stuff while (condition) };`  but is also, something I don't think it's blocking :)